common/types.ts

```diff
@@ -153,6 +153,109 @@ export interface INEUPrereqCourse {
   missing?: true;
 }
 
+//                                       NEW MAJOR OBJECT HERE
+/**
+ * A Major, containing all the requirements.
+ * @param name The name of the major.
+ * @param requirementSections A list of the sections of requirements.
+ * @param totalCreditsRequired Total credits required to graduate with this major.
+ * @param yearVersion The catalog version year of this major.
+ * @param concentrations The possible concentrations within this major.
+ */
+export interface Major2 {
+  name: string;
+  requirementSections: Section[];
+  totalCreditsRequired: number;
+  yearVersion: number;
+  concentrations: Concentrations;
+}
+
+/**
+ * A Section, containing its related requirements.
+ * @param title The title of the section.
+ * @param requirements A list of the requirements within this section.
+ * @param minRequirementCount The minimum number of requirements (counts from requirements)
+ *        that are accepted for the section to be fulfilled.
+ */
+export interface Section {
+  title?: string;
```

Comment:
lets make this title: string | null

Reply:
Let's not make this optional, since I'm pretty sure every section needs a title. I updated the docs, sorry for the confusion!